programs/fuzz/fuzz_privkey.c

```diff
@@ -0,0 +1,64 @@
+#include <stdint.h>
+#include "mbedtls/pk.h"
+
+//4 Kb should be enough for every bug ;-)
+#define MAX_LEN 0x1000
+
+
+int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+#ifdef MBEDTLS_PK_PARSE_C
+    int ret;
+    mbedtls_pk_context pk;
+
+    if (Size > MAX_LEN) {
+        //only work on small inputs
+        Size = MAX_LEN;
+    }
+
+    mbedtls_pk_init( &pk );
+    ret = mbedtls_pk_parse_key( &pk, Data, Size, NULL, 0 );
+    if (ret == 0) {
+#if defined(MBEDTLS_RSA_C)
+        if( mbedtls_pk_get_type( &pk ) == MBEDTLS_PK_RSA )
+        {
+            mbedtls_mpi N, P, Q, D, E, DP, DQ, QP;
+            mbedtls_rsa_context *rsa;
+
+            mbedtls_mpi_init( &N ); mbedtls_mpi_init( &P ); mbedtls_mpi_init( &Q );
+            mbedtls_mpi_init( &D ); mbedtls_mpi_init( &E ); mbedtls_mpi_init( &DP );
+            mbedtls_mpi_init( &DQ ); mbedtls_mpi_init( &QP );
+
+            rsa = mbedtls_pk_rsa( pk );
+            mbedtls_rsa_export( rsa, &N, &P, &Q, &D, &E );
```

Comment:
We expect that not only the call `mbedtls_rsa_export` will not trigger any bad behavior that ASan or UBSan detects, but also that it returns 0. Can this code be instrumented to check the return value? What should it do if it detects that an invariant has been breached? Call `exit()`? Call `abort()`? Call `LLVMFuzzerSomething()`?

Reply:
Yes it can
Call `abort` is the right way to go (I think it produces a signal caught by libFuzzer)
Here is an example (elliptic curve crypto differential fuzzing) : https://github.com/catenacyber/elliptic-curve-differential-fuzzer/blob/master/fuzz_ec.c#L195

Do you want a PR for that ?
Any news on #2748 ?

Reply:
Thanks for the reply! A PR would be appreciated if you have time.

For #2748, thank you for submitting this. I'm afraid none of us has had the time to dig into how the corpus is generated, and we don't want to merge something we don't understand, even if it's only test code. But we definitely appreciate your help and we do intend to get around to reviewing and merging it.

Reply:
Pushed #2999 (nice number)

For #2478, it could be renamed as it now goes beyond corpus generation for client and server targets. It also includes extending the tests so that fuzz targets are run on their corpuses with the sanitizers.
Anyways, let me know if you need help on it

Reply:
Thanks. We'll review #2478 as soon as we can, but I can't give you a date.